### PR TITLE
Dx alex

### PR DESCRIPTION
**Pull request by josojo**
current newest smart contract codes with some workflow tests.

This should be the latest version until Dominic's code with the storage logic is present.

### CODE REVIEW

**Review by W3stside — APPROVED** [review]
Tests run for me after rebase. Good to me

**Comment by W3stside** [comment]
Do not attempt to rebase this, just a simple merge commit

**Review by Velenir — APPROVED** [review]
Tests work great for me

File: test/dutchExchange.js [hunk 1]
```diff
@@ -155,7 +166,7 @@ contract('DutchExchange', (accounts) => {
       0,
       2,
       1,
-      { from: seller1 }
+      { from: seller1 },
```
Comment (josojo) [inline test/dutchExchange.js]: @W3stside Does the code run for you, if you have the comma here? My compiler always complains about it. That is why I removed it, although the linter says, it should be there.

**Pull request merged**